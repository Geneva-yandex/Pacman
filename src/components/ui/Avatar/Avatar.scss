.Avatar {
  display: inline-flex;
  border-radius: 50%;
  background-color: #535353;
  background-position: center;
  background-size: cover;
}
