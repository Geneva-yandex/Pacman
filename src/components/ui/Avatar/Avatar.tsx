import React from 'react';
import bem from 'easy-bem';
import './Avatar.scss';

interface IAvatarProps {
    src: string;
    size?: number;
}

const b = bem('Avatar');

const Avatar = ({src, size = 48}: IAvatarProps) => {
    const style = {
        backgroundImage: `url(${src})`,
        width: `${size}px`,
        height: `${size}px`
    };

    return <div className={b()} style={style}/>;
};

export default Avatar;
