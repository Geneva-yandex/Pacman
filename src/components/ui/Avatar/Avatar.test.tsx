import * as React from 'react';
import renderer from 'react-test-renderer';
import Avatar from './Avatar';

const props = {
    src: '/test.jpg',
    size: 30
};

describe('Avatar', () => {
    test('Avatar component renders correctly', () => {
        const renderedComponent = renderer
            .create(<Avatar {...props} />)
            .toJSON();

        expect(renderedComponent).toMatchSnapshot();
    });
});

