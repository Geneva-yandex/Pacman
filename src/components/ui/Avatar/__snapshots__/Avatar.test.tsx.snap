// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Avatar component renders correctly 1`] = `
<div
  className="Avatar"
  style={
    Object {
      "backgroundImage": "url(/test.jpg)",
      "height": "30px",
      "width": "30px",
    }
  }
/>
`;
